pkg/engine/datasource/graphql_datasource/graphql_datasource.go

```diff
@@ -1279,48 +1292,60 @@ func (s *Source) compactAndUnNullVariables(input []byte) []byte {
 		_ = json.Compact(buf, variables)
 		variables = buf.Bytes()
 	}
+
+	removeNullVariables := httpclient.IsInputFlagSet(input, httpclient.UNNULLVARIABLES)
+	variables = s.cleanupVariables(variables, removeNullVariables)
+
+	input, _ = jsonparser.Set(input, variables, "body", "variables")
+	return input
+}
+
+func (s *Source) cleanupVariables(variables []byte, removeNullVariables bool) []byte {
 	cp := make([]byte, len(variables))
```

Comment:
if second arg is false, we can skip this completely, no?

Reply:
This method is also cleaning up empty objects `{}`

so without directive we skip removing only top level variables with `null` value, but still removing `{}`